application.cpp

```diff
@@ -327,14 +329,33 @@ bool application::is_quiting() const {
    return my->_is_quiting;
 }
 
+//#define APPBASE_LOG_TIME
+#ifdef APPBASE_LOG_TIME
+long long log_time( const char* desc, long long start, int priority ) {
+   namespace bc = boost::chrono;
+   auto now = bc::duration_cast<bc::microseconds>( bc::system_clock::now().time_since_epoch() ).count();
+   auto time = now - start;
+   if( time > 5000 && start != 0) {
+      std::cerr << time << ' ' << desc << " p: " << priority << std::endl;
```

Comment:
Use of cerr has started to proliferate in appbase.    Consider adding registration of a logging function and using that.